### PR TITLE
test: restrict memory usage, so tests don't fail on travis

### PR DESCRIPTION
**Pull request by panchenko**
The changes from https://discuss.gradle.org/t/gradle-travis-sudo-false-leads-to-an-error/17034/3

With these numbers it works, it's a good question if it would be better to have numbers higher or lower :-)

### CODE REVIEW

**Comment by artembilan** [comment]
@panchenko ,

It is always pleasure to have contributions from you, but would you mind elaborating more what has made you to propose this change? 
And it is really good practice to include "Why?" to the commit message: https://chris.beams.io/posts/git-commit/

Thanks

**Comment by panchenko** [comment]
I've simplified changes here.

Lately tests in travis fail with exit code 137, in various internet topics that's explained as processed was killed by the container environment most likely because of high memory usage.

Currently [default max heap](https://docs.oracle.com/javase/8/docs/technotes/guides/vm/gc-ergonomics.html) is "Smaller of 1/4th of the physical memory or 1GB" and seems like that's too much for travis.

The only question if this limit should be used always or only on travis.

**Comment by garyrussell** [comment]
Thanks @panchenko - my inclination is to make it travis only - I (personally) have never had problems on my laptop.

**Comment by artembilan** [comment]
We use locally daemon and `--parallel` to build as fast as possible.
The `fork` is expensive operation because it starts a new JVM, so, that is really should be just for Travis.
Thanks

File: build.gradle [hunk 1]
```diff
@@ -150,6 +151,10 @@ subprojects { subproject ->
 			append = false
 			destinationFile = file("$buildDir/jacoco.exec")
 		}
+		if (isTravisBuild) {
+			// restrict memory usage, so tests don't fail with exit code 137 on travis
+			maxHeapSize = '256m'
```
Comment (artembilan) [inline build.gradle]: @panchenko ,

Looks like now we adjust only `Xmx` .
Isn't that Travis option really enough:
```
env:
  global:
  - MALLOC_ARENA_MAX=1
  - GRADLE_OPTS="-Xmx512m"
```
?

Comment (panchenko) [inline build.gradle]: There are 2 different things: the main gradle process which does the build, it's affected by GRADLE_OPTS and another forked process which performs the tests (currently it's forked once and performs all the tests, if specifying the "forkEvery" parameter then fork terminates after that number of tests and a new one is forked).

Currently only the forked "Gradle Test Executor" experience problems, so this change is enough.

Comment (artembilan) [inline build.gradle]: Fair enough, @panchenko !

Let me "put in on board" and merge!

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/4c130b02f02dda003a055d61ffdd5b12a89b963e

Thank you again, @panchenko !

**Pull request closed without merging**